dependencies/http/http_test.go

```diff
@@ -51,12 +54,23 @@ func TestLimitedDefaultClient(t *testing.T) {
 	})
 }
 
+// The TestValidator will let everything through _except_ a specific url, which
+// is redirected to from a http test server.
+type TestValidator struct{}
+
+func (TestValidator) Validate(anUrl *url.URL) error {
+	if anUrl.Host == "test-validator.example.com" {
+		return errors.New(codes.Invalid, "url is not valid, it connects to a private IP")
```

Comment:
nit but this error message is now a bit misleading, maybe change it to `test validator error`?